### ISSUE TITLE
SubtypeResolver is not properly copied when ObjectMapper.copy() is used

### ISSUE THREAD

**Issue by merzbird** [FasterXML/jackson-databind#2836]
When `ObjectMapper` is copied via `ObjectMapper.copy()`, the copy of `SubtypeResolver` is not properly propagated to all places, such as `SerializationConfig`/`DeserializationConfig`.

**Version information**
2.11.1, most likely introduced with https://github.com/FasterXML/jackson-databind/issues/2755
2.11.0 seems working fine.

**To Reproduce**
```
import com.fasterxml.jackson.databind.ObjectMapper;
public class JacksonSubtypeResolverIssue {
    public static void main(String[] args) {
        var objectMapper = new ObjectMapper();
        System.out.println(objectMapper.getSubtypeResolver() == objectMapper.getSerializationConfig().getSubtypeResolver());
        var copyObjectMapper = objectMapper.copy();
        System.out.println(copyObjectMapper.getSubtypeResolver() == copyObjectMapper.getSerializationConfig().getSubtypeResolver());
    }
}
```
Output: 
```
true
false
```

**Comment by cowtowncoder** [comment]
Is this same as #2785 (fixed in 2.11.2)?

**Comment by merzbird** [comment]
@cowtowncoder, yes, thanks, totally missed that new version with fix has already been released.
